task/dao.py

```diff
@@ -0,0 +1,167 @@
+#skipping dependency for CRUD operations (02/18)
+from db_helper import createTagField
+from db_classes import Task, TaskType, FieldValue, Field, Workspace
+
+
+def create(name, description, workspace_id, tags, due_date):
+    """ 
+    Create a new Task
+    Currently only supports creating a default Task
+    """
+
+    # TODO in later versions: split out getting a task type into a seperate helper function
+    # get the default task type for creating a default task
+    default_task_type = TaskType.objects(name = "Default")
+    default_task_type = default_task_type[0]
+    
+    # TODO in later versions: split out getting a Field into a seperate function
+    # get the fields for creating field values of the default task type
+    name_field = Field.objects(name = "Name")
+    name_field = name_field[0]
+
+    description_field = Field.objects(name = "Description")
+    description_field = description_field[0]
+
+    tag_fields = []
+    for tag in tags:
+        field = Field.objects(name = tag)
+        if len(field) == 0:
+            tag_field = createTagField(tag)
+            tag_fields.append(tag_field)
+        else:
+            # TODO in later versions: what to do if multiple fields of the same name exist. ie. one is a tag and another is a field
+            tag_fields.append(field[0])
+
+    due_date_field = Field.objects(name = "Due Date")
+    due_date_field = due_date_field[0]
+
+    # TODO in later versions: split out creating Field Values into a seperate helper function
+    # create non-static field values for the new task
+    ns_field_values = []
+    
+    name_field_value = FieldValue(value = name, task_type = None, field = name_field, allowed_value = None)
```

Comment:
This is totally not an issue, but for the record I think that if we don't pass a value for a (mongoengine) field and it isn't required then it will default to `None`, so we can probably get away with something like this:

```suggestion
    name_field_value = FieldValue(value=name, field=name_field)
```